### Year 3 Topology 4 - Limit Checking - Overloading, Voltage limits

In [1]:
import pandas as pd
import altair as alt

##### Loading

##### Percentage Loading in Branches 

Calculation related can be found in PSSE PAGVI Manual Chapter 6 Power system network simulations - Section 6.6.7 Activity Rate 

%I = MVA(actual)/(MVA(rated)*Vpu)

Vpu -> Bus Voltage at from end 

In [2]:
list_all = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for lsc in list_lsc:
            file_csv_out = 'savnw_sol_' + str(gen) +'_hy_' +str(gen_hy) +'_' + str(lsc)+'_l.csv'
            data_load = pd.read_csv(file_csv_out,header=1, skiprows=0)
            # Extracted data contains branch loading from the metered and non-metered ends. The data is filtered to obtain the branch loading from the metered end.
            data_fil = data_load
            data_fil['From Bus V'] = data_fil['From Bus Ext'].str.extract(r'(\d+\.\d+)') 
            data_fil['To Bus V'] = data_fil['To Bus Ext'].str.extract(r'(\d+\.\d+)')
            data_fil['Scenario']= 'Solar = ' + str(gen) + ' MW, ' +'Hydro = ' +str(gen_hy) +' MW, ' + lsc.upper()
            list_all.append(data_fil)
data_all = pd.concat(list_all).reset_index(drop=True)
data_all = data_all[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID',
       'Non Metered Bus', ' Branch I', 'Branch Flow MW', 'Branch Flow MVAR',
       'Branch MVA', 'MW Loss', 'MVAR Loss', 'Rate', '%I', '%MVA',
       'From Bus V', 'To Bus V']]

##### Line and Transformer Loading

Transformer and branch loading are categorised to following brackets based on the range to which it falls into 
* Loading less than 25%
* Loading 25% to 50%
* Loading 50% to 70%
* Loading 70% to 80%
* Loading 80% to 85%
* Loading 85% to 90%
* Loading 90% to 95%
* Loading 95% to 100%
* Loading 100% and above

Separating transformer and line branch loadings

In [3]:
transformer_loading = data_all[data_all['From Bus V']!=data_all['To Bus V']].reset_index(drop=True)
branch_loading = data_all[data_all['From Bus V'] == data_all['To Bus V']].reset_index(drop=True)

Defining loading brackets for transformer and line branches 

In [4]:
data = transformer_loading[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID','Branch Flow MW', 'Branch Flow MVAR','Branch MVA', 'Rate','%MVA']]
data['Loading'] = ''
data.loc[(data['%MVA'] >100), 'Loading'] = '%MVA>100'
data.loc[(data['%MVA'] <100), 'Loading'] = '95<%MVA<100'
data.loc[(data['%MVA'] <95), 'Loading'] = '90<%MVA<95'
data.loc[(data['%MVA'] <90), 'Loading'] = '85<%MVA<90'
data.loc[(data['%MVA'] <85), 'Loading'] = '80<%MVA<85'
data.loc[(data['%MVA'] <80), 'Loading'] = '70<%MVA<80'
data.loc[(data['%MVA'] <70), 'Loading'] = '50<%MVA<70'
data.loc[(data['%MVA'] <50), 'Loading'] = '25<%MVA<50'
data.loc[(data['%MVA'] <25), 'Loading'] = '%MVA<25'

<ipython-input-4-7f16dd71236d>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['Loading'] = ''


In [5]:
data_l = branch_loading[['Scenario','From Bus', 'From Bus Ext', 'To Bus', 'To Bus Ext', 'Ckt ID','Branch Flow MW', 'Branch MVA', 'Rate','%I']]
data_l['Loading'] = ''
data_l.loc[(data_l['%I'] >100), 'Loading'] = '%I>100'
data_l.loc[(data_l['%I'] <100), 'Loading'] = '95<%I<100'
data_l.loc[(data_l['%I'] <95), 'Loading'] = '90<%I<95'
data_l.loc[(data_l['%I'] <90), 'Loading'] = '85<%I<90'
data_l.loc[(data_l['%I'] <85), 'Loading'] = '80<%I<85'
data_l.loc[(data_l['%I'] <80), 'Loading'] = '70<%I<80'
data_l.loc[(data_l['%I'] <70), 'Loading'] = '50<%I<70'
data_l.loc[(data_l['%I'] <50), 'Loading'] = '25<%I<50'
data_l.loc[(data_l['%I'] <25), 'Loading'] = '%I<25'

<ipython-input-5-abcf08d21c83>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_l['Loading'] = ''


##### Histogram Transformer Branch Loading 

A histogram for each scenario is plotted to check the no of transformer branches in the above defined loading bracket 

In [6]:
alt.Chart(data).mark_bar().encode(
    alt.Y('Loading',scale = alt.Scale(domain=['%MVA>100','95<%MVA<100','90<%MVA<95','85<%MVA<90','80<%MVA<85','70<%MVA<80','50<%MVA<70','25<%MVA<50','%MVA<25'])),
    alt.X('count()')).properties(height = 120, width = 250).facet('Scenario',columns=5).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that for most of scenarios the transformer branch loading falls into a bracket of loading between 25 to 50 percentage. 

##### Histogram Line Branch Loading 

A histogram is plotted to check the no of line branches in the above defined loading bracket 

In [7]:
alt.Chart(data_l).mark_bar().encode(
    alt.Y('Loading', scale =alt.Scale(domain=['%I>100','95<%I<100','90<%I<95','85<%I<90','80<%I<85','70<%I<80','50<%I<70','25<%I<50','%I<25'])),
    alt.X('count()')).properties(height = 120, width = 250).facet('Scenario',columns=5).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that for most of scenarios the line branch loading falls into a bracket of loading between 25% to 50%

In the following section the lines and transformer branches with loading greater than 80% is filtered 

##### Checking transformer loading greater than 80% 

In [8]:
transformer_overload  = transformer_loading[transformer_loading['%MVA'] > 80.0].reset_index(drop=True)
transformer_overload['Branch'] = transformer_overload['From Bus'].astype(str) + ' ' + transformer_overload['From Bus Ext'] + ' to ' + \
transformer_overload['To Bus'].astype(str) + ' '+ transformer_overload['To Bus Ext'] + ' ' + transformer_overload['Ckt ID'].astype(str)
list_branch = transformer_overload['Branch'].unique()
print(list_branch)

['3018 CATDOG_G    13.800 to 3008 CATDOG      230.00 1']


In [9]:
transformer_overload

Scenario  From Bus        From Bus Ext  To Bus  \
0     Solar = 0 MW, Hydro = 400 MW, LLS      3018  CATDOG_G    13.800    3008   
1     Solar = 0 MW, Hydro = 400 MW, RLS      3018  CATDOG_G    13.800    3008   
2     Solar = 0 MW, Hydro = 400 MW, HLS      3018  CATDOG_G    13.800    3008   
3     Solar = 0 MW, Hydro = 500 MW, LLS      3018  CATDOG_G    13.800    3008   
4     Solar = 0 MW, Hydro = 500 MW, RLS      3018  CATDOG_G    13.800    3008   
5     Solar = 0 MW, Hydro = 500 MW, HLS      3018  CATDOG_G    13.800    3008   
6     Solar = 0 MW, Hydro = 600 MW, LLS      3018  CATDOG_G    13.800    3008   
7     Solar = 0 MW, Hydro = 600 MW, RLS      3018  CATDOG_G    13.800    3008   
8     Solar = 0 MW, Hydro = 600 MW, HLS      3018  CATDOG_G    13.800    3008   
9    Solar = 50 MW, Hydro = 400 MW, LLS      3018  CATDOG_G    13.800    3008   
10   Solar = 50 MW, Hydro = 400 MW, RLS      3018  CATDOG_G    13.800    3008   
11   Solar = 50 MW, Hydro = 400 MW, HLS      3018  CATDOG_G    13.800    3008   
12   Solar = 50 MW, Hydro = 500 MW, LLS      3018  CATDOG_G    13.800    3008   
13   Solar = 50 MW, Hydro = 500 MW, RLS      3018  CATDOG_G    13.800    3008   
14   Solar = 50 MW, Hydro = 500 MW, HLS      3018  CATDOG_G    13.800    3008   
15   Solar = 50 MW, Hydro = 600 MW, LLS      3018  CATDOG_G    13.800    3008   
16   Solar = 50 MW, Hydro = 600 MW, RLS      3018  CATDOG_G    13.800    3008   
17   Solar = 50 MW, Hydro = 600 MW, HLS      3018  CATDOG_G    13.800    3008   
18  Solar = 100 MW, Hydro = 400 MW, LLS      3018  CATDOG_G    13.800    3008   
19  Solar = 100 MW, Hydro = 400 MW, RLS      3018  CATDOG_G    13.800    3008   
20  Solar = 100 MW, Hydro = 400 MW, HLS      3018  CATDOG_G    13.800    3008   
21  Solar = 100 MW, Hydro = 500 MW, LLS      3018  CATDOG_G    13.800    3008   
22  Solar = 100 MW, Hydro = 500 MW, RLS      3018  CATDOG_G    13.800    3008   
23  Solar = 100 MW, Hydro = 500 MW, HLS      3018  CATDOG_G    13.800    3008   
24  Solar = 100 MW, Hydro = 600 MW, LLS      3018  CATDOG_G    13.800    3008   
25  Solar = 100 MW, Hydro = 600 MW, RLS      3018  CATDOG_G    13.800    3008   
26  Solar = 100 MW, Hydro = 600 MW, HLS      3018  CATDOG_G    13.800    3008   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
0   CATDOG      230.00       1             3018       4779            90.0   
1   CATDOG      230.00       1             3018       4793            90.0   
2   CATDOG      230.00       1             3018       4806            90.0   
3   CATDOG      230.00       1             3018       4783            90.0   
4   CATDOG      230.00       1             3018       4798            90.0   
5   CATDOG      230.00       1             3018       4819            90.0   
6   CATDOG      230.00       1             3018       4788            90.0   
7   CATDOG      230.00       1             3018       4803            90.0   
8   CATDOG      230.00       1             3018       4863            90.0   
9   CATDOG      230.00       1             3018       4777            90.0   
10  CATDOG      230.00       1             3018       4791            90.0   
11  CATDOG      230.00       1             3018       4804            90.0   
12  CATDOG      230.00       1             3018       4781            90.0   
13  CATDOG      230.00       1             3018       4796            90.0   
14  CATDOG      230.00       1             3018       4808            90.0   
15  CATDOG      230.00       1             3018       4785            90.0   
16  CATDOG      230.00       1             3018       4800            90.0   
17  CATDOG      230.00       1             3018       4847            90.0   
18  CATDOG      230.00       1             3018       4775            90.0   
19  CATDOG      230.00       1             3018       4790            90.0   
20  CATDOG      230.00       1             3018       4802            90.0   
21  CATDOG      230.00       1             301

##### Transformer branch overload plot

In [10]:
alt.Chart(transformer_overload).mark_line().encode(
    alt.X('%MVA', axis=alt.Axis(format='.3f'),scale=alt.Scale(zero=False)),
    alt.Y('Scenario'),
    alt.Color('Branch', legend=None)).properties(height = 300, width =400).facet('Branch',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Checking branch loading greater than 80% 

In [11]:

branch_overload = branch_loading[branch_loading['%I'] > 80.0].reset_index(drop=True)
branch_overload['Branch'] = branch_overload['From Bus'].astype(str) + ' ' + branch_overload['From Bus Ext'] + ' to ' + \
branch_overload['To Bus'].astype(str) + ' '+ branch_overload['To Bus Ext'] + ' ' + branch_overload['Ckt ID'].astype(str)
list_branch = branch_overload['Branch'].unique()
print(list_branch)


['154 DOWNTN      230.00 to 203 EAST230     230.00 1'
 '203 EAST230     230.00 to 154 DOWNTN      230.00 1']


In [12]:
branch_overload

Scenario  From Bus        From Bus Ext  To Bus  \
0     Solar = 0 MW, Hydro = 400 MW, RLS       154  DOWNTN      230.00     203   
1     Solar = 0 MW, Hydro = 400 MW, HLS       154  DOWNTN      230.00     203   
2     Solar = 0 MW, Hydro = 400 MW, HLS       203  EAST230     230.00     154   
3     Solar = 0 MW, Hydro = 500 MW, LLS       154  DOWNTN      230.00     203   
4     Solar = 0 MW, Hydro = 500 MW, LLS       203  EAST230     230.00     154   
5     Solar = 0 MW, Hydro = 500 MW, RLS       154  DOWNTN      230.00     203   
6     Solar = 0 MW, Hydro = 500 MW, RLS       203  EAST230     230.00     154   
7     Solar = 0 MW, Hydro = 500 MW, HLS       154  DOWNTN      230.00     203   
8     Solar = 0 MW, Hydro = 500 MW, HLS       203  EAST230     230.00     154   
9     Solar = 0 MW, Hydro = 600 MW, LLS       154  DOWNTN      230.00     203   
10    Solar = 0 MW, Hydro = 600 MW, LLS       203  EAST230     230.00     154   
11    Solar = 0 MW, Hydro = 600 MW, RLS       154  DOWNTN      230.00     203   
12    Solar = 0 MW, Hydro = 600 MW, RLS       203  EAST230     230.00     154   
13    Solar = 0 MW, Hydro = 600 MW, HLS       154  DOWNTN      230.00     203   
14    Solar = 0 MW, Hydro = 600 MW, HLS       203  EAST230     230.00     154   
15   Solar = 50 MW, Hydro = 500 MW, LLS       154  DOWNTN      230.00     203   
16   Solar = 50 MW, Hydro = 500 MW, LLS       203  EAST230     230.00     154   
17   Solar = 50 MW, Hydro = 500 MW, RLS       154  DOWNTN      230.00     203   
18   Solar = 50 MW, Hydro = 500 MW, RLS       203  EAST230     230.00     154   
19   Solar = 50 MW, Hydro = 500 MW, HLS       154  DOWNTN      230.00     203   
20   Solar = 50 MW, Hydro = 500 MW, HLS       203  EAST230     230.00     154   
21   Solar = 50 MW, Hydro = 600 MW, LLS       154  DOWNTN      230.00     203   
22   Solar = 50 MW, Hydro = 600 MW, LLS       203  EAST230     230.00     154   
23   Solar = 50 MW, Hydro = 600 MW, RLS       154  DOWNTN      230.00     203   
24   Solar = 50 MW, Hydro = 600 MW, RLS       203  EAST230     230.00     154   
25   Solar = 50 MW, Hydro = 600 MW, HLS       154  DOWNTN      230.00     203   
26   Solar = 50 MW, Hydro = 600 MW, HLS       203  EAST230     230.00     154   
27  Solar = 100 MW, Hydro = 500 MW, LLS       154  DOWNTN      230.00     203   
28  Solar = 100 MW, Hydro = 500 MW, RLS       154  DOWNTN      230.00     203   
29  Solar = 100 MW, Hydro = 500 MW, RLS       203  EAST230     230.00     154   
30  Solar = 100 MW, Hydro = 500 MW, HLS       154  DOWNTN      230.00     203   
31  Solar = 100 MW, Hydro = 500 MW, HLS       203  EAST230     230.00     154   
32  Solar = 100 MW, Hydro = 600 MW, LLS       154  DOWNTN      230.00     203   
33  Solar = 100 MW, Hydro = 600 MW, LLS       203  EAST230     230.00     154   
34  Solar = 100 MW, Hydro = 600 MW, RLS       154  DOWNTN      230.00     203   
35  Solar = 100 MW, Hydro = 600 MW, RLS       203  EAST230     230.00     154   
36  Solar = 100 MW, Hydro = 600 MW, HLS       154  DOWNTN      230.00     203   
37  Solar = 100 MW, Hydro = 600 MW, HLS       203  EAST230     230.00     154   

            To Bus Ext  Ckt ID  Non Metered Bus   Branch I  Branch Flow MW  \
0   EAST230     230.00       1              203        403         -148.20   
1   EAST230     230.00       1              203        410         -151.11   
2   DOWNTN      230.00       1              203        402          152.16   
3   EAST230     230.00       1              203        430         -160.66   
4   DOWNTN      230.00       1              203        423          161.82   
5   EAST230     230.00       1              203        439         -164.20   
6   DOWNTN      230.00       1              203        432          165.40   
7   EAST230     230.00       1              203        448         -166.97   
8   DOWNTN      230.00       1              203        441          168.23   
9   EAST230     230.00       1              203        467         -176.61   
10  DOWNTN   

Transmission line branch overload plot

In [13]:
alt.Chart(branch_overload).mark_line().encode(
    alt.X('%I', scale=alt.Scale(zero=False)),
    alt.Y('Scenario', sort='-x'),
    alt.Color('Branch', legend=None)).properties(height = 300, width =400).facet('Branch',columns=3).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Bus Voltage

In [14]:
list_volt = []
list_gens = [0,50,100]
list_lsc = ['LLS','RLS','HLS']
list_gen_hydro = [400,500,600] #dry, average, wet hydological scenarios
for gen in list_gens:
    for gen_hy in list_gen_hydro:
        for lsc in list_lsc:
            file_sav_in = 'savnw_sol_' + str(gen) +'_hy_' +str(gen_hy) +'_' + str(lsc)+'_volt.csv'
            data_volt = pd.read_csv(file_sav_in,header=1, skiprows=0)
            data_volt['Scenario']= 'Solar = ' + str(gen) + ' MW, ' +'Hydro = ' +str(gen_hy) +' MW, ' + lsc.upper()
            list_volt.append(data_volt)
data_all_volt = pd.concat(list_volt).reset_index(drop=True)
data_all_volt = data_all_volt[['Scenario','Bus Number', 'Bus Name', 'Area', 'Base Voltage', 'Bus Voltage(PU)','Bus Voltage(kV)']]

Similar to branch loading categorization into brackets of loadings, voltages are also categorised into brackets

In [15]:
data_v = data_all_volt
data_v['Voltage Range'] = ''
data_v.loc[(data_v['Bus Voltage(PU)'] >1.1), 'Voltage Range'] = 'Vbus>1.1'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.1), 'Voltage Range'] = '1.05<Vbus<1.1'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.05), 'Voltage Range'] = '1.0<Vbus<1.05'
data_v.loc[(data_v['Bus Voltage(PU)'] <1.0), 'Voltage Range'] = '0.95<Vbus<1.0'
data_v.loc[(data_v['Bus Voltage(PU)'] <0.95), 'Voltage Range'] = '0.95<Vbus<0.9'
data_v.loc[(data_v['Bus Voltage(PU)'] <0.9), 'Voltage Range'] = 'Vbus<0.9'

##### Histogram Bus Voltages

A histogram is plotted to check the no of Buses in each of the above defined brackets 

In [16]:
alt.Chart(data_v).mark_bar().encode(
    alt.Y('Voltage Range', scale =alt.Scale(domain=['Vbus>1.1','1.05<Vbus<1.1','1.0<Vbus<1.05','0.95<Vbus<1.0','0.95<Vbus<0.9','Vbus<0.9'])),
    alt.X('count()')).properties(height = 120, width = 250).facet('Scenario',columns=5).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

It can be seen that most of the buses have voltages in the bracket of 1.0 PU and 1.05 PU

In the following section, the buses out of upper and lower normal operating range is filtered and reported 

##### Voltage Upper Limits 

In [17]:
data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)['Bus Number'].unique()

array([3018,  206,  211,  212], dtype=int64)

In [18]:
volt_high = data_all_volt[data_all_volt['Bus Voltage(PU)']>1.05].reset_index(drop=True)
volt_high

Scenario  Bus Number      Bus Name  Area  \
0     Solar = 0 MW, Hydro = 400 MW, LLS        3018  CATDOG_G         5   
1     Solar = 0 MW, Hydro = 400 MW, RLS        3018  CATDOG_G         5   
2     Solar = 0 MW, Hydro = 500 MW, LLS        3018  CATDOG_G         5   
3     Solar = 0 MW, Hydro = 500 MW, HLS         206  URBGEN           2   
4     Solar = 0 MW, Hydro = 600 MW, LLS        3018  CATDOG_G         5   
5     Solar = 0 MW, Hydro = 600 MW, RLS         206  URBGEN           2   
6     Solar = 0 MW, Hydro = 600 MW, HLS         211  HYDRO_G          2   
7     Solar = 0 MW, Hydro = 600 MW, HLS         212  HYDRO_N          2   
8    Solar = 50 MW, Hydro = 400 MW, LLS        3018  CATDOG_G         5   
9    Solar = 50 MW, Hydro = 400 MW, RLS        3018  CATDOG_G         5   
10   Solar = 50 MW, Hydro = 500 MW, LLS        3018  CATDOG_G         5   
11   Solar = 50 MW, Hydro = 500 MW, HLS         206  URBGEN           2   
12   Solar = 50 MW, Hydro = 600 MW, LLS        3018  CATDOG_G         5   
13  Solar = 100 MW, Hydro = 400 MW, LLS        3018  CATDOG_G         5   
14  Solar = 100 MW, Hydro = 400 MW, RLS        3018  CATDOG_G         5   
15  Solar = 100 MW, Hydro = 500 MW, LLS        3018  CATDOG_G         5   
16  Solar = 100 MW, Hydro = 500 MW, RLS        3018  CATDOG_G         5   
17  Solar = 100 MW, Hydro = 500 MW, HLS         206  URBGEN           2   
18  Solar = 100 MW, Hydro = 600 MW, LLS        3018  CATDOG_G         5   

    Base Voltage  Bus Voltage(PU)  Bus Voltage(kV)  Voltage Range  
0           13.8            1.054           14.547  1.05<Vbus<1.1  
1           13.8            1.051           14.502  1.05<Vbus<1.1  
2           13.8            1.053           14.534  1.05<Vbus<1.1  
3           18.0            1.053           18.946  1.05<Vbus<1.1  
4           13.8            1.052           14.520  1.05<Vbus<1.1  
5           18.0            1.053           18.949  1.05<Vbus<1.1  
6           20.0            1.052           21.046  1.05<Vbus<1.1  
7           20.0            1.052           21.046  1.05<Vbus<1.1  
8           13.8            1.055           14.552  1.05<Vbus<1.1  
9           13.8            1.051           14.508  1.05<Vbus<1.1  
10          13.8            1.054           14.540  1.05<Vbus<1.1  
11          18.0            1.055           18.987  1.05<Vbus<1.1  
12          13.8            1.053           14.527  1.05<Vbus<1.1  
13          13.8            1.055           14.557  1.05<Vbus<1.1  
14          13.8            1.052           14.513  1.05<Vbus<1.1  
15          13.8            1.054           14.545  1.05<Vbus<1.1  
16          13.8            1.051           14.500  1.05<Vbus<1.1  
17          18.0            1.053           18.956  1.05<Vbus<1.1  
18          13.8            1.053           14.532  1.05<Vbus<1.1

##### Upper voltage limit violation plot

In [19]:
alt.Chart(volt_high).mark_line().encode(
    alt.X('Bus Voltage(PU)', axis=alt.Axis(format='.3f'), scale=alt.Scale(nice=False, zero=False)),
    alt.Y('Scenario', sort='-x'),
    alt.Color('Bus Number:N', legend=None)).properties(height = 150, width = 250).facet('Bus Number:N',columns=2).resolve_scale(x='independent', y = 'independent')

alt.FacetChart(...)

##### Voltage Lower Limits

In [20]:
data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95]['Bus Number'].unique()

array([], dtype=int64)

In [21]:
data_all_volt[data_all_volt['Bus Voltage(PU)']<0.95]

Empty DataFrame
Columns: [Scenario, Bus Number, Bus Name, Area, Base Voltage, Bus Voltage(PU), Bus Voltage(kV), Voltage Range]
Index: []

#### Conclusion 

Year 3 Toplogy 4 limit violation check for overloading and voltage was carried out for the three load scenarios. It was seen that
1. Overloading of greater than 80% was reported for the branch ' 154 DOWNTN      230.00 to 203 EAST230     230.00 1'. and no transformers were overloaded for considered load scenarios. 
   For the branch ' 154 DOWNTN      230.00 to 203 EAST230     230.00 1' the scenarios of overload > 80% are
    * Solar = 0 MW, Hydro = 400 MW, RLS, HLS
    * Solar = 0 MW, Hydro = 500 MW, LLS, RLS, HLS
    * Solar = 0 MW, Hydro = 600 MW, LLS, RLS, HLS
    * Solar = 50 MW, Hydro = 500 MW, LLS, RLS, HLS
    * Solar = 50 MW, Hydro = 600 MW, LLS, RLS, HLS
    * Solar = 100 MW, Hydro = 500 MW, LLS, RLS, HLS
    * Solar = 100 MW, Hydro = 600 MW, LLS, RLS, HLS
2. The buses violating the upper voltage limit for various scenarios are:
    * Solar = 0 MW, Hydro = 500 MW, HLS	
      206	URBGEN
    * Solar = 0 MW, Hydro = 600 MW, HLS	
      211	HYDRO_G, 212	HYDRO_N	
    * Solar = 50 MW, Hydro = 500 MW, HLS	
      206	URBGEN
    * Solar = 100 MW, Hydro = 500 MW, HLS
      206	URBGEN